File: infrastructure/src/main/java/com/study/platform/global/kafka/KafkaMessagePublisherAdapter.java
package com.study.platform.global.kafka;

import lombok.RequiredArgsConstructor;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

import java.util.concurrent.CompletableFuture;

@Component
@RequiredArgsConstructor
public class KafkaMessagePublisherAdapter implements KafkaMessagePublisher {

    private final KafkaTemplate<String, String> kafkaTemplate;

    @Override
    public CompletableFuture<Void> publish(String topic, String key, String payload) {
        return kafkaTemplate.send(topic, key, payload).thenApply(result -> null);
    }
}

File: infrastructure/src/main/kotlin/com/study/platform/global/kafka/KafkaMessagePublisherAdapter.kt
package com.study.platform.global.kafka

import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component
import java.util.concurrent.CompletableFuture

@Component
class KafkaMessagePublisherAdapter(
    private val kafkaTemplate: KafkaTemplate<String, String>
) : KafkaMessagePublisher {

    override fun publish(topic: String, key: String?, payload: String): CompletableFuture<Void> =
        if (key != null) kafkaTemplate.send(topic, key, payload).thenApply { null }
        else kafkaTemplate.send(topic, payload).thenApply { null }
}
